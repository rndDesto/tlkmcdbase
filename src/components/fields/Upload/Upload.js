import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scoped.css';

export default function Upload(props) {
  const { className, input, inputProps, data, label, helper, meta } = props;
  const { dirty, error, touched } = meta;
  const classes =   [
    styles.root,
    (data === null || (data && !data.name)) && styles.placeholder,
    !!error && (dirty || touched) && styles.error,
    className
  ].filter(Boolean).join(' ');
  return (
    <div className={classes}>
      <label>{label}</label>
      <div className={classes}>
        <label>
          <p>{data !== null && data.name ? data.name  : 'Belum ada file dipilih'}</p>
          <p>Upload</p>
          <input  id={input.name} {...input} {...inputProps} type="file" />
        </label>
      </div>
      {!!error && (dirty || touched) && <small className={styles.error}>{error}</small>}
      <span>{helper}</span>
    </div>
  );
}
Upload.defaultProps = {
  className: '',
  data:{},
  helper: '',
  input: {},
  inputProps: {},
  label: '',
  meta:{},
};

Upload.propTypes = {
  className: PropTypes.string,
  data: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.string
  ]),
  helper: PropTypes.string,
  input: PropTypes.object,
  inputProps: PropTypes.object,
  label: PropTypes.string,
  meta: PropTypes.object,
};
