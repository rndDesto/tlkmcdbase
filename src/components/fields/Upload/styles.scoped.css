.root > label {
  color: var(--grey-color-1);
  font-size: 0.875rem;
  line-height: 1rem;
  font-weight: bold;
}

.root > div > label {
  display: grid;
  grid-template-columns: 1fr 0fr;
}

.root > div > label > input[type="file"] {
  display: none;
}

.root > div > label > p {
  padding: 7px;
}

.root > div > label > p:nth-child(2) {
  align-items: center;
  background: var(--primary-color);
  border-radius: 3px;
  color: var(--background-color-2);
  display: flex;
  justify-content: center;
  padding: 7px 16px;
  width: fit-content;

}

.root > div {
  border: 0.0625rem solid var(--light-text-color);
  border-radius: 0.25rem;
  box-shadow: none;
  color: var(--text-color);
  display: block;
  font-size: 0.875rem;
  font-weight: bold;
  margin-bottom: 0.5rem;
  margin-top: 0.625rem;
  padding: 0.4rem;
  width: 100%;
  transition: border 0.25s;
}

.root.placeholder > label > p:first-child  {
  color: var(--light-text-color);
}

.root > .error {
  border-color: var(--error-color);
}

.root + small.error {
  color: var(--error-color);
  display: block;
  font-size: 0.75rem;
}
