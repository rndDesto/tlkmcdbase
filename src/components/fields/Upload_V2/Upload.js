import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scoped.css';

export default function Upload(props) {
  const { className, input, inputProps, data, label, helper, meta } = props;
  const { dirty, error, touched } = meta;
  const classes =   [
    styles.root,
    !!data || styles.placeholder,
    !!error && (dirty || touched) && styles.error,
    className
  ].filter(Boolean).join(' ');
  return (
    <div className={classes}>
      <label>{label}</label>
      <div className={classes}>
        <label>
          <p>{data ? data.name : 'Belum ada file dipilih'}</p>
          <div>
            <img alt="logee" src="../../../assets/ic-upload.svg" />
          </div>
          <input  id={input.name} {...input} {...inputProps} type="file"/>
        </label>
      </div>
      {!!error && (dirty || touched) && <small className={styles.error}>{error}</small>}
      <span>{helper}</span>
    </div>
  );
}
Upload.defaultProps = {
  className: '',
  data:{},
  helper: '',
  input: {},
  inputProps: {},
  label: '',
  meta:{},
};

Upload.propTypes = {
  className: PropTypes.string,
  data: PropTypes.object,
  helper: PropTypes.string,
  input: PropTypes.object,
  inputProps: PropTypes.object,
  label: PropTypes.string,
  meta: PropTypes.object,
};
