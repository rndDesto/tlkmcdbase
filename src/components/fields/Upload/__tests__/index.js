import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Upload from '../Upload';

jest.mock('react', () => ({
  ...require.requireActual('react'),
  useEffect: cb => cb(),
}));

describe('src/components/fields/Upload', () => {
  test('render', () => {
    global.URL.createObjectURL = jest.fn();
    document.getElementById = () => ({ src: null });

    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Upload  />);
    expect(tree).toMatchSnapshot();

    const res = Upload({ ...Upload.defaultProps, data: { name: 'a' } });
    expect(res).toMatchSnapshot();

    const res1 = Upload({
      ...Upload.defaultProps,
      meta: { error: 'err', dirty: true, touched: true }
    });
    expect(res1.props.children[2].props.children).toBe('err');

    const res2 = Upload({
      ...Upload.defaultProps,
      meta: { error: 'err', dirty: false, touched: true }
    });
    expect(res2.props.children[2].props.children).toBe('err');
  });
});
