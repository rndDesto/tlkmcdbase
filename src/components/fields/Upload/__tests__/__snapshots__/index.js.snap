// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/fields/Upload render 1`] = `
<div
  className="true"
>
  <label>
    
  </label>
  <div
    className="true"
  >
    <label>
      <p />
      <p>
        Upload
      </p>
      <input
        type="file"
      />
    </label>
  </div>
  <span>
    
  </span>
</div>
`;

exports[`src/components/fields/Upload render 2`] = `
<div
  className="true"
>
  <label>
    
  </label>
  <div
    className="true"
  >
    <label>
      <p>
        a
      </p>
      <p>
        Upload
      </p>
      <input
        type="file"
      />
    </label>
  </div>
  <span>
    
  </span>
</div>
`;
